pkg/controller/petset/pet.go

```diff
@@ -63,6 +67,65 @@ type pcb struct {
 	parent *apps.StatefulSet
 }
 
+//Gets the number of replicas stored in pod's StatefulSetReplicasAnnotation. If an the annotation
+//is not present, or if it contains an invalid value, -1 is returned. Otherwise, a 32 bit integer
+//representation of the annotation's value is returned.
+func getReplicas(pod *api.Pod) int32 {
+	if pod == nil {
+		return -1
+	} else if replicaStr, exists := pod.Annotations[StatefulSetReplicasAnnotation]; !exists {
+		glog.Errorf("Pod %s has no annotation for %s", pod.Name, StatefulSetReplicasAnnotation)
+		return -1
+	} else if replicas64, err := strconv.ParseInt(replicaStr, 10, 32); err != nil {
+		glog.Errorf("Pod %s has invalid value %s for annotation %s",
+			pod.Name, replicaStr, StatefulSetReplicasAnnotation)
+		return -1
+	} else {
+		return int32(replicas64)
+	}
+}
+
+//Sets pod's StatefulSetReplicasAnnotation to the base 10 string value of replicas.
+func setReplicas(p *api.Pod, replicas int32) {
+	p.Annotations[StatefulSetReplicasAnnotation] = strconv.FormatInt(int64(replicas), 10)
+}
+
+//Updates pod to be equivalent to desired, and to contain the number same number of replicas as set.
+//By equivalent, we mean that all of the IdentityMappers for StatefulSet produce equivalent values
+//for both pod and desired. The returned bool is true if pod has been mutated or false if no
+//mutation occurs. If the returned error is not nil, the returned bool is false.
+func update(set *apps.StatefulSet, id string, pod *api.Pod, desired *api.Pod) (bool, error) {
+	//preconditions
+	if set == nil {
+		return false, errors.New("nil StatefulSet parameter")
+	} else if pod == nil {
+		return false, errors.New("nil source supplied")
+	} else if desired == nil {
+		return false, errors.New("nil target supplied")
+	}
```

Comment:
The presence of this code seems to imply that we expect that sometimes it will be called this way, but as I understand it, this shouldn't happen. If we had to have it, I think we could just check that all 3 are not nil in a single statement and return an error.